src/main/java/net/minestom/server/listener/SpectateListener.java

```diff
@@ -13,11 +13,22 @@ public static void listener(ClientSpectatePacket packet, Player player) {
         final Entity target = Entity.getEntity(targetUuid);
 
         // Check if the target is valid
-        if (target == null || target == player)
+        if (target == null || target == player) {
             return;
+        }
 
-        // TODO check if 'target' is in a different instance
-        player.spectate(target);
+        if (target.getInstance() == null || player.getInstance() == null) {
+            return;
+        }
+
+        if (target.getInstance().getUniqueId() != player.getInstance().getUniqueId()) {
```

Comment:
Won't this always return?

Reply:
If they're in the same instance they'll bypass this check, i see no reason for it to always return.